### PR TITLE
refactor: api-code-refactor and aws-rds-configuration

### PR DESCRIPTION
**Pull request by pavle0903**

### CODE REVIEW

File: backend/src/main/java/com/example/userservice/controller/RegisteredUserController.java [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
+@AllArgsConstructor
 public class RegisteredUserController implements UserServiceApi {
-    @Autowired
-    private RegisteredUserService userService;
+
+    private final RegisteredUserService userService;
 
     @PostMapping("/addUser")
-    public RegisteredUser addUser(@RequestBody RegisteredUser registeredUser){
+    public RegisteredUserEntity addUser(@RequestBody RegisteredUserEntity registeredUser){
```
Comment (ilicjelena) [inline backend/src/main/java/com/example/userservice/controller/RegisteredUserController.java]: Use DTO object (RegisteredUser) for communication between frontend and backend. Do not use entity object here. Do the transformation in RegisteredUserService.

File: backend/src/main/java/com/example/userservice/controller/RegisteredUserController.java [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
         return userService.addUser(registeredUser);
     }
 
     @Override
     @GetMapping("/{id}")
-    public com.example.userservice.api.RegisteredUser getUser(Integer id) {
-        RegisteredUser user = userService.getUser(id);
-        return new com.example.userservice.api.RegisteredUser(user.getEmail(), user.getName(), user.getSurname());
+    public RegisteredUser getUser(Integer id) {
+        return userService.getUser(id);
```
Comment (ilicjelena) [inline backend/src/main/java/com/example/userservice/controller/RegisteredUserController.java]: Check will it work without @PathVariable annotation for id?

File: backend/src/main/java/com/example/userservice/model/RegisteredUserEntity.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
     @Column(name="phone", nullable=false)
     private String phone;
 
-    public RegisteredUser(String email, String password, String name, String surname, String phone) {
+    public RegisteredUserEntity(String email, String password, String name, String surname, String phone) {
```
Comment (ilicjelena) [inline backend/src/main/java/com/example/userservice/model/RegisteredUserEntity.java]: better to use @AllArgsConstructor :)

File: backend/src/main/java/com/example/userservice/service/RegisteredUserServiceImpl.java [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
-    @Autowired
-    private IRegisteredUserRepository userRepository;
+
+    private final RegisteredUserRepository userRepository;
+    private final ModelMapper modelMapper;
 
     @Override
-    public RegisteredUser addUser(RegisteredUser registeredUser) {
+    public RegisteredUserEntity addUser(RegisteredUserEntity registeredUser) {
         return  userRepository.save(registeredUser);
```
Comment (mzorboski) [inline backend/src/main/java/com/example/userservice/service/RegisteredUserServiceImpl.java]: Service should return DTO object, not Entity

File: backend/src/main/java/com/example/userservice/model/RegisteredUserEntity.java (hunk 3 above)
Comment (mzorboski) [inline backend/src/main/java/com/example/userservice/model/RegisteredUserEntity.java]: Is it needed to have this constructor? Can we just use lombok?

**Review by ilicjelena — APPROVED** [review]

**Pull request merged**